src/PlayersManager.ts

```diff
@@ -659,28 +659,41 @@ export class PlayersManager extends Dispatcher {
         }
       }
       // we got here - this is not a defaultPath - we need to find the relevant hotspot and see if it has startFrom
-      const currentNodeHotspots = this.raptData.hotspots.filter(
-        (hotspot) => hotspot.nodeId === currentNode.id
+      if (!clickedHotspotId) {
+        log(
+            "error",
+            "pm_checkSeekTo",
+            `Error: hotspot with id: ${clickedHotspotId} is undefined. returning startFrom=0`
+        );
+        return 0;
+      }
+
+      const clickedHotspot = this.raptData.hotspots.find(
```

Comment:
just to verify that `this.raptData.hotspots` is always available and cannot be undefined.

if so, ignore this comment

Reply:
yes, if there isn't then the code will fail much earlier. also, before my fix the code was using `this.raptData.hotspots` to create a list of all hotspots of the current node. see line [662](https://github.com/kaltura/kaltura-interactive-player/blob/master/src/PlayersManager.ts#L662)
so i think we're safe.